### PR TITLE
Update stdsimd to undo an accidental stabilization

### LINKED ISSUE #52403: Update stdsimd submodule to destabilize an accidental stabilization
This is a tracking issue for updating `stdsimd` to pull in https://github.com/rust-lang-nursery/stdsimd/pull/524, destabilizing a currently nightly-only API. Want to make sure we don't forget about this by accident!

### PR DESCRIPTION
**Pull request by alexcrichton**
Closes #52403

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r=Mark-Simulacrum

carrying over approval from https://github.com/rust-lang/rust/pull/52429

**Comment by alexcrichton** [comment]
[This error](https://github.com/rust-lang/rust/pull/52433#issuecomment-405519179) was suspected to be caused by this update, so I'm running a test locally

**Comment by kennytm** [comment]
The same powerpc error in https://github.com/rust-lang/rust/pull/52570#issuecomment-406645793, but again it is a rollup 🤷.

**Comment by alexcrichton** [comment]
@bors: r-

I'll take a look

**Comment by alexcrichton** [comment]
@gnzlbg as a heads up I had to update this to stdsimd master to pull in https://github.com/rust-lang-nursery/stdsimd/pull/531 which also pulls in the deletion of `std::simd`

@bors: r=Mark-Simulacrum

**Comment by gnzlbg** [comment]
@alexcrichton the std::simd replacement is already fully compatible with
what we had before so people can just use that. Let me know if you want to
pull it into the nursery.

**Comment by gnzlbg** [comment]
@alexcrichton 

> oh I thought I recalled you saying you didn't want to delete std::simd just yet while you waited to publish your first version, but I may be misremembering! Other than that it sounds great!

You are remembering it correctly, but I changed my mind about doing releases. Those who want to use this will need to add a `git` dependency on it, which ties them to nightly, and forbids them from releasing stuff. 

That's unfortunate, but it will prevent the issue that we had with `faster` where the compiler moved on, and `faster` and all their dependencies stopped working because the released versions of `coresimd` became unsupported. 

> If you'd like I can move it to the nursery as well

I think that would be best.

**Comment by alexcrichton** [comment]
@gnzlbg ok, if you want to catch me on IRC I can help you transfer the repo to the nursery

**Comment by gnzlbg** [comment]
I’m travelling this weekend and connectivity is low, but I think this can
wait till Monday.

File: src/libcore/lib.rs [hunk 1]
```diff
@@ -244,9 +244,6 @@ macro_rules! vector_impl { ($([$f:ident, $($args:tt)*]),*) => { $($f!($($args)*)
 #[cfg(not(stage0))] // allow changes to how stdsimd works in stage0
 mod coresimd;
 
-#[unstable(feature = "stdsimd", issue = "48556")]
-#[cfg(not(stage0))]
-pub use coresimd::simd;
```
Comment (pitdicker) [inline src/libcore/lib.rs]: This removal broke the CI of rand, but I have trouble finding the reason. Is there an alternative to use simd with `no_std`, or is this no longer supported?

**Comment by nnethercote** [comment]
perf.rust-lang.org indicates that this caused a [significant compile-time win](https://perf.rust-lang.org/compare.html?start=f8f6e7c04d1331574589df298e8651e7104fe2ff&end=17eb392cefdcf4b98c47f9ab36742e66a314932f&stat=instructions:u), which is (a) nice, and (b) surprising to me.

@alexcrichton, do you know why this might have improved rustc speed?

**Comment by gnzlbg** [comment]
@nnethercote we removed a very big module from `stdsimd`: https://github.com/rust-lang-nursery/stdsimd/pull/528 but I don't know how could that make `rustc` compile things faster.

**Comment by alexcrichton** [comment]
Yes this removed a large unstable module from libcore. I believe it had a **ton** of trait impls, so maybe metadata loading for libstd is faster now? It could mean perhaps that metadata loading isn't as efficient as it could otherwise be perhaps?

**Comment by gnzlbg** [comment]
> I believe it had a ton of trait impls,

This is correct. It also had a ton of types with tons of methods. 

FYI what was removed now lives in the nursery: https://github.com/rust-lang-nursery/packed_simd

**Comment by MoSal** [comment]
FYI, while it's obviously removed from the nightly documentation, `std::simd` is still in the official documentation [here](https://doc.rust-lang.org/std/simd/index.html).

This may confuse some people (e.g. game developers) who are googling "Rust SIMD".

@steveklabnik

**Comment by gnzlbg** [comment]
That's weird, the APIs are unstable so they should only appear on the nightly docs, but those are not in `nightly/`... :/

**Comment by MoSal** [comment]
@gnzlbg 

Unstable APIs appear in stable docs. Maybe they shouldn't, but they do. And maybe it's too late to change that behavior.

**Pull request merged**